WeatherMonitoringSystem/Domain/Utilities/UserInteraction.cs

```diff
@@ -0,0 +1,111 @@
+﻿
+using WeatherMonitoringSystem.Bots;
+using WeatherMonitoringSystem.Core;
+using WeatherMonitoringSystem.DataParsers;
+
+namespace WeatherMonitoringSystem.UI;
+
+public static class UserInteraction
+{
+        public static void Run()
+    {
+        PrintMessage("Welcome to the Real-Time Weather Monitoring System!");
+
+        WeatherDataParserFactory parserFactory = new WeatherDataParserFactory();
+
+        while (true)
+        {
+            string format = GetDataFormat();
+
+            try
+            {
+                parserFactory.SetParser(format);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                continue;
+            }
+
+            IWeatherDataParser parser = parserFactory.GetParser();
+            WeatherData weatherData = GetWeatherData(parser);
+            if (weatherData == null) return;
+
+            ActivateBots(weatherData);
+
+            Console.WriteLine("Do you want to change the parsing method? (yes/no)");
+            string change = Console.ReadLine().Trim().ToLower();
+            if (change != "yes") break;
+        }
+    }
+
+    private static string GetDataFormat()
+    {
+        return GetUserInput("Enter data format (JSON/XML): ").Trim().ToLower();
+    }
+
+    private static WeatherData GetWeatherData(IWeatherDataParser parser)
```

Comment:
since `WeatherDataParserFactory` should be a strategy and already have the parser, maybe this can be moved to that class , but we need to remove the print statements and keep the responsibility to the `UserInteraction`.